File: Temas/Geometria Computacional/task_5a/template_5a.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>


class Point {
public:
    int x;
    int y;
};

std::vector<int> determine_nearest_point(const std::vector<int>& S, const std::vector<int>& Q) {
    /*
    Given a set S of points on a line and a set Q of query points,
    find the closest neighbor of the query point,
    i.e., a point from S with the minimum distance to the query point.
    Return array of answers on the queries
    */
    std::vector<int> results;

    // YOUR CODE HERE

    return results;
}

int main() {
    int n;
    std::cin >> n;
    int m;
    std::vector<int> line_points(n);
    for (int &item : line_points) {
        std::cin >> item;
    }
    std::cin >> m;
    std::vector<int> queries(m);
    for (int i = 0; i < m; ++i) {
        std::cin >> queries[i];
    }
    std::vector<int> result;
    result = determine_nearest_point(line_points, queries);
    for (int i = 0; i < m; ++i) {
        std::cout << result[i] << std::endl;
    }
}

File: Temas/Geometria Computacional/task_5a/template_5a.py
import sys
import typing as tp


class Point:
    x: int
    y: int


def determine_nearest_point(S: tp.List[int], Q: tp.List[int]) -> tp.List[int]:
    """
    Given a set S of points on a line and a set Q of query points,
    find the closest neighbor of the query point,
    i.e., a point from S with the minimum distance to the query point.
    Return array of answers on the queries
    """
    results: tp.List[int] = list()

    # YOUR CODE HERE

    return results


if __name__ == '__main__':
    n = int(input())
    line_points = list(map(int, input().split()))
    m = int(input())
    queries: tp.List[int] = [int() for _ in range(m)]
    for i in range(0, m):
        queries[i] = int(input())

    result = determine_nearest_point(line_points, queries)
    for i in range(0, m):
        print(result[i])


